api/controllers/AddTreatmentController.js

```diff
@@ -5,51 +5,95 @@
  * @help        :: See https://sailsjs.com/docs/concepts/actions
  */
 
-const {
-  conditionReducer,
-  oneAttribute,
-} = require('../utils/condition.mapper');
-
-const dataStore = require('../utils/DataStore');
-const ConditionHelper = require('../utils/ConditionHelpers');
+const arrayHelpers = require('../utils/ArrayHelpers');
+const conditionReducer = require('../utils/ConditionHelpers').conditionReducer;
 
 module.exports = {
-  create: function (req, res) {
-    let medicalReport = req.session.medicalReport;
-    const conditionList = conditionReducer(medicalReport.conditions);
+  create: async function (req, res) {
+    // load the medical report
+    let medicalReport = await MedicalReport.findOne({
+      where: {
+        applicationCode: req.session.applicationCode
+      },
+      include: [
+        { model: Condition, as: 'Conditions' }
+      ]
+    });
+
+    // get a list suitable for checkbox list component
+    let conditionList = conditionReducer(medicalReport.Conditions);
+
+    // if there's only one condition, default to it being selected
+    if (Object.keys(conditionList).length === 1) {
+      _.set(res.locals, 'data.selectedConditions', conditionList);
+    }
 
     res.view('pages/treatments/add', {
       conditionList: conditionList,
-      oneValue: oneAttribute(conditionList),
       medicalReport: medicalReport,
     });
   },
 
-  store: function (req, res) {
-    const body = Object.assign({}, req.body);
-    delete body._csrf;
+  store: async function (req, res) {
+    // load the medical report
+    let medicalReport = await MedicalReport.findOne({
+      where: {
+        applicationCode: req.session.applicationCode
+      }
+    });
 
     // use the value of the submit button to determine redirect
-    const action = body.save_and;
-
-    /**
-     * If there are newConditions, create them pre-validation
-     * and auto-select them
-     */
-    if (body.newConditions) {
-      req.body = ConditionHelper.addConditions(req, body, 'treatmentTreatedCondition');
-    }
+    const action = req.body.save_and;
 
     let valid = req.validate(req, res, require('../schemas/treatment.schema'));
 
     if (valid) {
       // save model here
-      if (!_.has(req.session.medicalReport, 'treatments')) {
-        req.session.medicalReport.treatments = [];
+      let treatment = await Treatment.create({
```

Comment:
Requested change: For non-`newConditions`, we should see if they exist in the database. If they don't, it'd be good to send an error back (it might be an indication that someone is injecting something that could invalidate the data).

Reply:
That's what's happening here:
![image](https://user-images.githubusercontent.com/1187115/72844776-3d8c6600-3c6b-11ea-99ff-1a227368994f.png)

We take the array of IDs that come from the form, loop over them and query the database to get an instance of the Condition, then only if one comes back from the db, we attach it to the treatment. 

At this point, it fails silently, but it won't allow you to attach something that doesn't exist. 

We could add a further where clause to this to ensure that the condition belongs to the currently selected MedicalReport (will push that up momentarily)

After typing all of this, realize that you were specifically talking about trying to catch during validation.... above is still relevant. 

We can investigate some way of doing this in the validation step. Not really sure what that looks like tho.

Reply:
Just pushed this change which ensures the Condition belongs to the selected Medical Report:
![image](https://user-images.githubusercontent.com/1187115/72847020-b68dbc80-3c6f-11ea-9a21-709e8a1e513f.png)